x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -356,29 +357,34 @@ export const AlertsTableComponent: React.FC<AlertsTableComponentProps> = ({
 
   const leadingControlColumns = useMemo(() => getDefaultControlColumn(ACTION_BUTTON_COUNT), []);
 
+  const casesPermissions = useGetUserCasesPermissions();
+  const CasesContext = kibana.services.cases.getCasesContext();
+
   if (loading || indexPatternsLoading || isEmpty(selectedPatterns)) {
     return null;
   }
 
   return (
-    <StatefulEventsViewer
-      additionalFilters={additionalFiltersComponent}
-      currentFilter={filterGroup}
-      defaultCellActions={defaultCellActions}
-      defaultModel={alertsDefaultModel}
-      end={to}
-      entityType="events"
-      hasAlertsCrud={hasIndexWrite && hasIndexMaintenance}
-      id={timelineId}
-      leadingControlColumns={leadingControlColumns}
-      onRuleChange={onRuleChange}
-      pageFilters={defaultFiltersMemo}
-      renderCellValue={RenderCellValue}
-      rowRenderers={defaultRowRenderers}
-      scopeId={SourcererScopeName.detections}
-      start={from}
-      utilityBar={utilityBarCallback}
-    />
+    <CasesContext owner={[APP_ID]} userCanCrud={casesPermissions?.crud ?? false}>
```

Comment:
This is the important change. To show the create case flyout, and eventually the case modal, the cases context needs initialization. Effectively this is the place where the flyout renders when invoked.